src/VirtualClient/VirtualClient.Actions/MLPerfTraining/MLPerfTrainingExecutor.cs

```diff
@@ -0,0 +1,515 @@
+                return filter;
+            }
+
+            set
+            {
+                this.Parameters[nameof(MLPerfTrainingExecutor.DiskFilter)] = value;
+            }
+        }
+
+        /// <summary>
+        /// The MLPerf Training model name (e.g. bert, rnnt, ssd-mobilenet).
+        /// </summary>
+        public string Model
+        {
+            get
+            {
+                return this.Parameters.GetValue<string>(nameof(this.Model));
+            }
+        }
+
+        /// <summary>
+        /// The current running user
+        /// </summary>
+        public string Username
+        {
+            get
+            {
+                string username = this.Parameters.GetValue<string>(nameof(MLPerfTrainingExecutor.Username));
+                if (string.IsNullOrWhiteSpace(username))
+                {
+                    username = this.GetCurrentUserName(true);
+                }
+
+                return username;
+            }
+        }
+
+        /// <summary>
+        /// ML specific parameter used to control size of training samples
+        /// </summary>
+        public string BatchSize
+        {
+            get
+            {
+                return this.Parameters.GetValue<string>(nameof(this.BatchSize));
+            }
+        }
+
+        /// <summary>
+        /// The framework used for the implementation. It will be a sub directory in the model/implementations directory
+        /// </summary>
+        public string Implementation
+        {
+            get
+            {
+                return this.Parameters.GetValue<string>(nameof(this.Implementation));
+            }
+        }
+
+        /// <summary>
+        /// Container image name
+        /// </summary>
+        public string ContainerName
+        {
+            get
+            {
+                return this.Parameters.GetValue<string>(nameof(this.ContainerName));
+            }
+        }
+
+        /// <summary>
+        /// Name of zip file without .zip. It will be the name of the folder where the model package is extracted to
+        /// </summary>
+        public string DataPath
+        {
+            get
+            {
+                return this.Parameters.GetValue<string>(nameof(this.DataPath));
+            }
+        }
+
+        /// <summary>
+        /// Number of GPUs to be utilized
+        /// </summary>
+        public string NumberOfGPUs
+        {
+            get
+            {
+                return this.Parameters.GetValue<string>(nameof(this.NumberOfGPUs));
+            }
+        }
+
+        /// <summary>
+        /// Config file for GPUs
+        /// </summary>
+        public string ConfigFile
+        {
+            get
+            {
+                return this.Parameters.GetValue<string>(nameof(this.ConfigFile));
+            }
+        }
+
+        /// <summary>
+        /// The MLPerf Training Nvidia code directory.
+        /// </summary>
+        protected string NvidiaDirectoryPath
```

Comment:
Please make the name clear and match the description. Also let's remove Nvidia in the name if we want to reuse this class for AMD GPU later

Reply:
The [repository](https://github.com/mlcommons/training_results_v2.1) is structured in a way that the training results include submissions by multiple companies, and their trial to run the benchmarks. Here, we are using the implementation of the models, submitted by NVIDIA as the training models as such are an unstable release. The folder doesn't consider the GPU provider. In the future, if we have AMD GPUs, their support and implementations will be a sub-directory inside the NVIDIA directory.